gpiozero/output_devices.py

```diff
@@ -535,109 +536,123 @@ class TonalBuzzer(SourceMixin, CompositeDevice):
-            plural = 's' if octaves > 1 else ''
-            raise ValueError(
-                'mid_note too low for {} octave{}'.format(octaves, plural))
-
-    def note_value(self, note):
-        """
-        Convert the given note to a normalized value scaled -1 to 1 relative to
-        the buzzer's range. If note is ``None``, ``None`` is returned.
-        """
-        if note is None:
-            return None
-        note = self._note_to_midi(note)
-        return (note - self.mid_note) / (self.max_note - self.mid_note)
+            return super(TonalBuzzer, self).__repr__()
 
-    def play(self, note):
+    def play(self, tone):
         """
-        Play the given note e.g. ``play(60)`` (MIDI) or ``play('C4')`` (note).
-        ``play(None)`` turns the buzzer off.
+        Play the given *tone*. This can either be an instance of
+        :class:`~gpiozero.tones.Tone` or can be anything that could be used to
+        construct an instance of :class:`~gpiozero.tones.Tone`.
+
+        For example::
+
+            >>> from gpiozero import TonalBuzzer
+            >>> from gpiozero.tones import Tone
+            >>> b = TonalBuzzer(17)
+            >>> b.play(Tone("A4"))
+            >>> b.play(Tone(220.0)) # Hz
+            >>> b.play(Tone(60)) # middle C in MIDI notation
+            >>> b.play("A4")
+            >>> b.play(220.0)
+            >>> b.play(60)
         """
-        if note is None:
+        if tone is None:
             self.value = None
         else:
-            freq = self._note_to_freq(note)
-            if self.min_frequency <= freq <= self.max_frequency:
+            if not isinstance(tone, Tone):
+                tone = Tone(tone)
+            freq = tone.frequency
+            if self.min_tone.frequency <= tone <= self.max_tone.frequency:
                 self.pwm_device.pin.frequency = freq
                 self.pwm_device.value = 0.5
             else:
-                raise ValueError("note out of the device's range")
+                raise ValueError("tone is out of the device's range")
 
     def stop(self):
         """
         Turn the buzzer off. This is equivalent to setting :attr:`value` to
-        ``None``.
+        :data:`None`.
         """
         self.value = None
 
+    @property
+    def tone(self):
+        """
+        Returns the :class:`~gpiozero.tones.Tone` that the buzzer is currently
+        playing, or :data:`None` if the buzzer is silent. This property can
+        also be set to play the specified tone.
+        """
+        if self.pwm_device.pin.frequency is None:
+            return None
+        else:
+            return Tone.from_frequency(self.pwm_device.pin.frequency)
+
+    @tone.setter
+    def tone(self, value):
+        self.play(value)
+
     @property
     def value(self):
         """
         Represents the state of the buzzer as a value between 0 (representing
         the minimum note) and 1 (representing the maximum note). This can also
```

Comment:
Seems to be some confusion in the docstrings as to whether value is 0 -> 1 or -1 -> 1 ? :confused:

Reply:
It started as 0->1 but should be -1->1 now.

Reply:
Whoops - I'll fix that in a mo!